### DISCUSSION TITLE
Read, interprete and send A0 during booting

### DISCUSSION THREAD

**Discussion by Maschler** [arendst/Tasmota#18750] [Q&A] [answered]
Hello all,

I would like to implement the following with the Rules:

**First part: At Boot**
During the starting process of ESP, it should read the current state of the analog pin (A0). The value should then be interpreted directly. Afterward, the interpreted result should be published via MQTT. I tried:
_Rule1
ON System#Boot DO IF (analog#a0>500) publish Machine/STATE 1 ELSE publish Machine/STATE 0 ENDIF ENDON_

As Code:
`ON System#Boot DO IF (analog#a0>500) publish Machine/STATE 1 ELSE publish Machine/STATE 0 ENDIF ENDON`

The feedback from the Console is:
`Machine/RESULT = {"Command":"Unknown"}`

**Second part: Rest of time **
After booting, only interpreted changes to the value of the analog pin (A0) are to be sent via MQTT. The second part **works fine**
_Rule1
ON analog#a0>500 DO publish Machine/STATE 1 ENDON
ON analog#a0<=500 DO publish Machine/STATE 0 ENDON_

As Code:
`Rule1
ON analog#a0>500 DO publish Machine/STATE 1 ENDON
ON analog#a0<=500 DO publish Machine/STATE 0 ENDON`

**How can the analog value be transferred during booting?**

Thanks a lot!

**Comment by sfromis** [comment]
You cannot use trigger expressions with `#` anywhere but in the trigger part of a rule, between `on` and `do`. And IF/ELSE/ENDIF is only available for a custom build to include this feature. "Normal" rule triggers are available shortly after boot, and your "second part" rules are not limited to changing values, but every update cycle, changed or not.

**Comment by Maschler** [comment]
Thanks for reply. 

How can I then read the current state of the A0 for the first time after booting and send it initially via MQTT?

Because if I don't do that, then the last MQTT state can be = 1 although currently the state is = 0.

**Edit:**
I see that the state is publish after Wifi is connected, but before the MQTT connection is established.
(Code: `Rule1 ON analog#a0>500 DO publish Machine/STATE 1 ENDON ON analog#a0<=500 DO publish Machine/STATE 0 ENDON`)

00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at FB, Count 153
00:00:00.055 QPC: Count 1
00:00:00.082 Project tasmota - Tasmota Version 12.5.0(tasmota)-2_7_4_9(2023-04-17T08:03:54)
00:00:00.551 WIF: Connecting to AP1 xxxx Channel 1 BSSId xx:xx:xx:xx:xx:xxin mode xx as Machine-6324...
00:00:02.759 WIF: Connected
00:00:03.012 HTP: Web server active on Machine-6324 with IP address 192.168.xxx.xxx
**00:00:04.483 RUL: ANALOG#A0<=500 performs "publish Machine/STATE 0"
00:00:04.487 RSL: STATE = 0
00:00:06.598 MQT: Attempting connection...
00:00:06.611 MQT: Connected**
00:00:06.614 MQT: Machine/LWT = Online (retained)
00:00:06.617 MQT: Machine/cmnd/POWER = 
00:00:06.626 MQT: Machine/INFO1 = {"Info1":xxxxxx}
00:00:06.641 MQT: Machine/INFO2 = {"Info2":xxxxxx}
00:00:06.656 MQT: Machine/INFO3 = {"Info3":xxxxxx}
00:00:06.667 MQT: Machine/RESULT = {"POWER1":"OFF"}
00:00:06.669 MQT: Machine/POWER1 = OFF

How can i solve that issue?

**Reply by sfromis** [reply]
Well, If I create a rule like `on analog#a0>2000 ...` it will trigger shortly after boot. I do not see why you even ask for a special thing during booting.

**Reply by Maschler** [reply]
The connection to MQTT is not established at this point.

**Comment by Maschler** [comment]
Is it possible to combine two triggers like this:

ON **((analog#a0>500) AND (Mqtt#Connected))** DO publish Machine/STATE 1 ENDON
ON **((analog#a0<500) AND (Mqtt#Connected))** DO publish Machine/STATE 0 ENDON

**Reply by barbudor** [reply, marked as answer]
No but you can save the value of A0 (from analog0 trigger) into a var
And then trigger on mqtt#connected to publish the value

Don't forget that a trigger is an event. It happens. It's not a state or a value
